### PR TITLE
make titiler tile endpoints consistent with tile cache service

### PR DESCRIPTION
**Pull request by solomon-negusse**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the new behavior?

Titiler endpoints use the same pattern as tile cache service

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/routes/titiler/integrated_alerts.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+
+# will turn this on when we're ready to replace tile cache service
+# @router.get(
+#     f"/{dataset}/{{version}}/dynamic/{{z}}/{{x}}/{{y}}.png",
+#     response_class=Response,
+#     tags=["Raster Tiles"],
+#     response_description="PNG Raster Tile",
+# )
+@router.get(
```
Comment (solomon-negusse) [inline app/routes/titiler/integrated_alerts.py]: the `/titiler` endpoint for integrated alerts here is temporary to deploy this and test it without replacing the existing dynamic implementation. there is an item in the roadmap to fully implement this later.

File: app/routes/titiler/algorithms/alerts.py [hunk 2]
```diff
@@ -20,44 +18,14 @@ class Alerts(BaseAlgorithm):
...
-    )
-
-    render_type: RenderType = Field(
-        RenderType.true_color,
-        description="Render true color or encoded pixels",
-    )
+    start_date: Optional[str] = None
+    end_date: Optional[str] = None
+    alert_confidence: Optional[str] = None
+    render_type: Optional[RenderType] = RenderType.true_color
```
Comment (danscales) [inline app/routes/titiler/algorithms/alerts.py]: Shouldn't render_type just be RenderType - not Optional[RenderType].  You've given it a default value, and it really should be true_color or encoded, not None, right?

File: app/main.py [hunk 3]
```diff
@@ -42,6 +42,8 @@
 from .routes import preview
 
 from .routes.titiler import routes as titiler_routes
+from .routes.titiler.integrated_alerts import router as alerts_router
```
Comment (danscales) [inline app/main.py]: Very minor: Maybe rename "alerts_router" to "integrated_alerts_router" for consistency with next line?

**Review by gtempus — APPROVED** [review]
@solomon-negusse and I walked through this together yesterday. I'm fine with the temporary `titiler` _implementation_. Moving away from the auto-_magic_ titiler endpoints is worth it, even though it's a little more work. The URLs are much cleaner 🧹 

Thanks, Solomon! 💪

File: app/routes/titiler/integrated_alerts.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    When using `latest` call will be redirected (307) to version tagged
+    as latest.
+    """
+
+    latest = "latest"
+
+
+_versions = get_versions(dataset, TileCacheType.cog)
+for _version in _versions:
+    extend_enum(GfwIntegratdAlertsVersions, _version, _version)
```
Comment (danscales) [inline app/routes/titiler/integrated_alerts.py]: Can you add comments on what GfwIntegratedAlertsVersions() is for?  I don't see it used anywhere, so then it seems like a lot of work to create an enum with all the integrated alerts versions going back for several years.

File: app/routes/titiler/umd_glad_dist_alerts.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+today = date.today()
+
+
+@router.get(
+    f"/{dataset}/{{version}}/dynamic/{{z}}/{{x}}/{{y}}.png",
+    response_class=Response,
+    tags=["Raster Tiles"],
+    response_description="PNG Raster Tile",
+)
+async def glad_dist_alerts_raster_tile(
```
Comment (danscales) [inline app/routes/titiler/umd_glad_dist_alerts.py:31]: I wonder if we should get rid of glad_ here and in the name of the file?  So file name is just umd_dist_alerts.py and name here is dist_alerts_raster_tile.  Would be more parallel with integrated_alerts.  Also, including glad may confuse with the glad-L and glad-S2 alerts.

**Review by danscales — APPROVED** [review]
Just a few comments/suggested improvements, but otherwise looks good.

File: app/routes/titiler/umd_glad_dist_alerts.py (hunk 5 above)
Comment (solomon-negusse) [reply app/routes/titiler/umd_glad_dist_alerts.py:31]: i added gfw_ prefix to integrated alerts.  `{org}_optional{group}` prefix is used consistently in other route files.  GLAD  is just the name of the UMD  research group.

File: app/routes/titiler/integrated_alerts.py (hunk 4 above)
Comment (solomon-negusse) [reply app/routes/titiler/integrated_alerts.py]: I have added back the use of that enum in the view function which will show to the user the available valid versions to use.

File: app/main.py (hunk 3 above)
Comment (solomon-negusse) [reply app/main.py]: Done

File: app/routes/titiler/algorithms/alerts.py (hunk 2 above)
Comment (solomon-negusse) [reply app/routes/titiler/algorithms/alerts.py]: Good catch, thanks.

**Comment by danscales** [comment]
Your new changes in response to some of my comments look good.  I guess you just have to figure out the test failure, then you can merge the PR.

**Pull request merged**